lib/utils/angular/scroll_host/angular_2.dart

```diff
@@ -0,0 +1,323 @@
+
+  @override
+  int get clientWidth => _scrollHost.clientWidth;
+
+  @override
+  num get offsetX => _scrollHost.offsetX;
+
+  @override
+  num get offsetY => _scrollHost.offsetY;
+
+  @override
+  Element get anchorElement => _scrollHost.anchorElement;
+
+  @override
+  Stream<ScrollHostEvent> get onScroll => _scrollHost.onScroll;
+
+  @override
+  PanController get panController => _scrollHost.panController;
+
+  @override
+  StickyController get stickyController => _scrollHost.stickyController;
+
+  @override
+  int get scrollLength => _scrollHost.scrollLength;
+
+  @override
+  int get scrollPosition => _scrollHost.scrollPosition;
+
+  @override
+  Stream<IntersectionObserverEntry> onIntersection(Element element) =>
+      _scrollHost.onIntersection(element);
+
+  @override
+  void ngOnDestroy() {
+    dispose();
+  }
+}
+
+/// Provides a scroll host that uses the browser window content area.
+@Injectable()
+class WindowScrollHost extends WindowScrollHostBase implements OnDestroy {
+  WindowScrollHost(DomService domService, NgZone ngZone,
+      GestureListenerFactory gestureListenerFactory, Window window)
+      : super(domService, ngZone, gestureListenerFactory, window);
+
+  @override
+  ngOnDestroy() {
+    dispose();
+  }
+}
+
+/// A directive to track already sticking Elements on the page (e.g. one that
+/// has position fixed).
+///
+/// The sticky controller will respect their position and
+/// will put the other sticky headers and footers below or above them.
+@Directive(
+  selector: '[acxStickyFloating]',
+)
+class StickyFloatingTracker implements OnInit, OnDestroy {
+  final ScrollHost _scrollHost;
+  final Element _element;
+
+  StickyFloatingTracker(this._scrollHost, ElementRef elementRef)
+      : _element = elementRef.nativeElement;
+
+  @override
+  void ngOnInit() {
+    _scrollHost.stickyController.trackFloating(_element);
+  }
+
+  @override
+  void ngOnDestroy() {
+    _scrollHost.stickyController.untrackFloating(_element);
+  }
+}
+
+/// A directive to dynamically set a class based on the pan state.
+///
+/// Example use:
+///
+///     <some-element acxPanClass="'pan-style'" ...
+///
+/// The directive will add -top, -right, -bottom and -left suffix to the style
+/// name, adding and removing it in a dom write callback.
+/// E.g. in the example above, if the panning is top-left, both the
+/// `pan-style-top` and the `pan-style-left` styles names will be set.
+@Directive(
+  selector: '[acxPanClass]',
+)
+class AcxPanClassDirective extends BasePanClassDirective
+    implements OnInit, OnDestroy {
+  AcxPanClassDirective(
+      DomService domService, ScrollHost scrollHost, ElementRef elementRef)
+      : super(domService, scrollHost, elementRef.nativeElement);
+
+  @override
+  void ngOnInit() => startPanListener();
+
+  @override
+  void ngOnDestroy() => stopPanListener();
+
+  // Overridden to add Input annotation and keep angular2 dep outside of base
```

Comment:
angular2 --> Angular

(Should this file be renamed as well, given that it is `angular_2.dart`?)